### PR TITLE
[DOC-2349] [PL-44156] Update gRPC requirement

### PR DESCRIPTION
**Pull request by brian-f-harness**
SMP is not using GRPC, all communication goes via REST now. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by smjt-h — APPROVED** [review]

File: docs/first-gen/fg-sme/k8s/existing-cluster-kubernetes-on-prem-infrastructure-requirements.md [hunk 1]
```diff
@@ -119,6 +119,11 @@ Follow the steps on creating the load balancer as part of the process described
 
 You must also open port 9879 on the load balancer and map it to port 9879 on the ingress controller. This is to support gRPC traffic.
 
+:::info note
+This update is not required for delegate version 23.12.818xx and later.
```
Comment (brian-f-harness) [inline docs/first-gen/fg-sme/k8s/existing-cluster-kubernetes-on-prem-infrastructure-requirements.md:123]:
```suggestion
This update is not required for delegate version 23.12.81803 and later.
```

File: docs/first-gen/fg-sme/vm/embedded-kubernetes-on-prem-infrastructure-requirements.md [hunk 2]
```diff
@@ -135,6 +135,11 @@ Typically, you will also set up DNS to resolve a domain to the frontend IP, and
 
 You also need to open port 9879 on the Load Balancer and map it to port 9879 on the Ingress controller. This is to support gRPC traffic.
 
+:::info note
+This update is not required for delegate version 23.12.818xx and later.
```
Comment (brian-f-harness) [inline docs/first-gen/fg-sme/vm/embedded-kubernetes-on-prem-infrastructure-requirements.md:139]:
```suggestion
This update is not required for delegate version 23.12.81803 and later.
```

File: docs/first-gen/firstgen-platform/account/manage-delegates/delegate-requirements-and-limitations.md [hunk 3]
```diff
@@ -60,8 +60,8 @@ The following network requirements are for connectivity between the Harness Dele
 All network connections from your local network to Harness SaaS are outbound-only.* HTTPS port 443 outbound from the Delegate to Harness.
 :::
 
-* HTTP/2 for gRPC (gRPC Remote Procedure Calls)
-* Delegate requirements: The Delegate will need API/SSH/HTTP access to the providers you add to Harness, such as:
+- HTTP/2 for gRPC (gRPC Remote Procedure Calls). This is not required for delegate version 23.12.818xx and later.
```
Comment (brian-f-harness) [inline docs/first-gen/firstgen-platform/account/manage-delegates/delegate-requirements-and-limitations.md:63]:
```suggestion
- HTTP/2 for gRPC (gRPC Remote Procedure Calls). This is not required for delegate version 23.12.81803 and later.
```

File: docs/first-gen/firstgen-platform/account/manage-delegates/delegate-requirements-and-limitations.md [hunk 4]
```diff
@@ -73,13 +73,18 @@ For more information, see [Supported Platforms and Technologies](../../../starth
 
 ### gRPC Limitations
 
+:::info note
+gRPC is not required for delegate version 23.12.818xx and later.
```
Comment (brian-f-harness) [inline docs/first-gen/firstgen-platform/account/manage-delegates/delegate-requirements-and-limitations.md:77]:
```suggestion
gRPC is not required for delegate version 23.12.81803 and later.
```

File: docs/platform/delegates/delegate-concepts/delegate-requirements.md [hunk 5]
```diff
@@ -67,10 +67,15 @@ All network connections from your local network to Harness SaaS are outbound-onl
 	+ Collaboration providers
 	+ SSH access to target physical and virtual servers
 
+
 ### gRPC limitations
 
-If you do not enable gRPC connections, the following limitations apply:
+:::info note
+gRPC connections are not required for delegate version 23.12.818xx and later.
```
Comment (brian-f-harness) [inline docs/platform/delegates/delegate-concepts/delegate-requirements.md:74]:
```suggestion
gRPC connections are not required for delegate version 23.12.81803 and later.
```

File: docs/self-managed-enterprise-edition/deploy-with-kots/kubernetes-cluster-on-prem-infrastructure-requirements.md [hunk 6]
```diff
@@ -149,9 +149,15 @@ The installation of Harness Self-Managed Enterprise Edition requires a load bala
 After Harness Self-Managed Enterprise Edition is installed, the load balancer is used to access the Harness Manager UI with a web browser.
 
 For information on how to create the load balancer, go to [Install Self-Managed Enterprise Edition with KOTS](kubernetes-cluster-on-prem-kubernetes-cluster-setup.md).
+ |
 
 ### gRPC and load balancer settings
 
+:::info note
+gRPC settings are not required for delegate version 23.12.818xx and later.
```
Comment (brian-f-harness) [inline docs/self-managed-enterprise-edition/deploy-with-kots/kubernetes-cluster-on-prem-infrastructure-requirements.md:157]:
```suggestion
gRPC settings are not required for delegate version 23.12.81803 and later.
```

File: docs/self-managed-enterprise-edition/deploy-with-kots/kubernetes-cluster-on-prem-kubernetes-cluster-setup.md [hunk 7]
```diff
@@ -394,6 +394,11 @@ You can change the settings later in the KOTS admin console **Config** tab:
 
 #### gRPC and load balancer settings
 
+:::info note
+gRPC settings are not required for delegate version 23.12.818xx and later.
```
Comment (brian-f-harness) [inline docs/self-managed-enterprise-edition/deploy-with-kots/kubernetes-cluster-on-prem-kubernetes-cluster-setup.md:398]:
```suggestion
gRPC settings are not required for delegate version 23.12.81803 and later.
```

**Review by xingchi-jin — APPROVED** [review]
approve for the docs/platform/delegates/delegate-concepts/delegate-requirements.md
and docs/first-gen/firstgen-platform/account/manage-delegates/delegate-requirements-and-limitations.md files

**Comment by brian-f-harness** [comment]
@arya-harness @akshay-khole could one of you please review for the SMP changes? Thanks.

**Pull request merged**